packages/permissionless/errors/account.ts

```diff
@@ -0,0 +1,306 @@
+                "• The initCode factory is creating an account, but is not implemented correctly as it is not deploying at the sender address",
+                "",
+                "Possible solutions:",
+                "• Verify that the factory address in the initCode is correct (the factory address is the first 20 bytes of the initCode).",
+                "• Verify that the initCode factory is implemented correctly. The factory must deploy the smart account at the sender address.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountNotDeployed extends BaseError {
+    static message = /aa20/
+    override name = "SmartAccountNotDeployed"
+    constructor({
+        cause,
+        sender,
+        docsPath
+    }: {
+        cause?: BaseError
+        sender: Address
+        docsPath?: string
+    }) {
+        super(
+            [
+                `Smart account ${sender} is not deployed.`,
+                "",
+                "Possible reasons:",
+                "• An initCode was not specified, but the sender address (i.e. the smart account) is not deployed.",
+                "",
+                "Possible solutions:",
+                "• If this is the first transaction by this account, make sure the initCode is included in the user operation.",
+                "• If the smart account is already supposed to be deployed, verify that you have selected the correct sender address for the user operation.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountDoNotHaveEnoughFunds extends BaseError {
+    static message = /aa21/
+    override name = "SmartAccountDoNotHaveEnoughFunds"
+    constructor({
+        cause,
+        sender,
+        docsPath
+    }: {
+        cause?: BaseError
+        sender: Address
+        docsPath?: string
+    }) {
+        super(
+            [
+                `You are not using a paymaster, but the ${sender} address did not have enough native tokens to cover the gas costs associated with the user operation.`,
+                "",
+                "Possible solutions:",
+                "• If you are not using a paymaster, verify that the sender address has enough native tokens to cover the required pre-fund. Consider leveraging functions like getRequiredPrefund.",
+                "• If you are looking to use a paymaster to cover the gas fees, verify that the paymasterAndData field is set.",
+                "",
+                docsPath ? `Docs: ${docsPath}` : ""
+            ].join("\n"),
+            {
+                cause
+            }
+        )
+    }
+}
+
+export class SmartAccountSignatureExpiredOrNotDue extends BaseError {
```

Comment:
SmartAccountSignatureExpiredOrNotDue -> SignatureExpiredOrNotDue?

Reply:
Similar reason, as we have similar error in Paymaster too